### ISSUE TITLE
[4.2.3]: validateBusinessTaxIdAsVatId is throwing an error instead of returning a validation error

### ISSUE THREAD

**Issue by WHITE-developer** [craftcms/commerce#3046]
### Description
When you have `validateBusinessTaxIdAsVatId` enabled and you update an address on the cart an `Invalid element – craft\errors\InvalidElementException` is thrown on this https://github.com/craftcms/commerce/blob/4.2.3/src/elements/Order.php#L2108 line

### Steps to reproduce

1. update your cart on the front-end with validateBusinessTaxIdAsVatId enabled and a random vatID

### Expected behavior
I would expect a normal validation error to occur that is getting returned with the `cart.errors`

### Actual behavior
an `InvalidElementException` is getting thrown and you are unable to finish your checkout

### Craft CMS version

4.3.3

### Craft Commerce version

4.2.3

### PHP version

8.1.6

### Operating system and version

Ubuntu 22.04

### Database type and version

mysql 8.0.30

**Comment by pdaleramirez** [comment]
@WHITE-developer I can't seem to replicate this issue with 

```
    'useBillingAddressForTax' => true,
    'validateBusinessTaxIdAsVatId' => true,
```
setting. Could you provide a video screencast of the error? or send us your DB on support@craftcms.com.

**Comment by pdaleramirez** [comment]
I am able to replicate this issue now. We'll update this ticket soon.

**Comment by pdaleramirez** [comment]
If you input an invalid `Organization Tax ID` in the Address Checkout Form it throws the Invalid element – craft\errors\InvalidElementException form

![Image](https://user-images.githubusercontent.com/4172750/212060909-87174732-bfcf-44dd-9906-519a4e7b9dc5.png)
![Image](https://user-images.githubusercontent.com/4172750/212060910-e1f62d3b-b936-4233-91b5-64a085490692.png)

**Comment by nikolenko-dmitriy** [comment]
I also faced this problem

![2023-02-01_14-58](https://user-images.githubusercontent.com/65535954/216050157-53da62a7-446a-4065-b267-8faaeb937026.png)

**Comment by nfourtythree** [comment]
Hi all

Thank you for bringing this to our attention.

We have pushed up a fix for this which will be included in the next release of Commerce.

Thanks!

**Comment by nfourtythree** [comment]
Fix in commit: https://github.com/craftcms/commerce/commit/1e628a93b4c5e7fb23648cc1b01aceeb467af195

**Comment by nikolenko-dmitriy** [comment]
@nfourtythree

> Fix in commit: [1e628a9](https://github.com/craftcms/commerce/commit/1e628a93b4c5e7fb23648cc1b01aceeb467af195)

This commit is a nice update to make field `organizationTaxId` to be validated on action `users/save-address`. I like it, thanks for this.

But the original trouble having the error `Element XXXX could not be duplicated because it doesn't validate` related to another thing.

It happens on the checkout page with action `commerce/cart/update-cart` trying to add the existing address associated with user's account to the actual order. As I can see, plugin creates a new address for the order each time when the form with action `commerce/cart/update-cart` is submitted with parameters `billingAddressId` or `shippingAddressId`. Even if the address information wasn't actually changed. That's why in the error message was mentioned `could not be duplicated`.

So, with current logic it's expected that on the checkout page where user might select already existing addresses, all them should have a valid field `organizationTaxId`. But on form submission the new validation is still happens when user selects the existing address from his account and it's duplicated and attached to the actual order with the new `ID`.

And it might be the case when during this new validation it fails and as the result the same error `Element XXXX could not be duplicated because it doesn't validate` is triggered. One of the reason of this case might be the situation when VIES webservice is down, what often happens to it. The related issue is here https://github.com/craftcms/commerce/issues/2500.

This need more attention because with such bug it's not possible to migrate the website to Craft 4 and Craft Commerce 4 for webshops where validation for `Organization Tax ID` is enabled.

### Additional info
- Craft CMS version: 4.3.6.1
- Craft Commerce version: 4.2.4
